opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -452,3 +452,19 @@ func WorkingPodForRollingRestart(ctx context.Context, k8sClient client.Client, s
 	}
 	return "", errors.New("unable to calculate the working pod for rolling restart")
 }
+
+// DeleteOSDDeployment deletes the OSD deployment along with all its pods
+func DeleteOSDDeployment(ctx context.Context, k8sClient client.Client, clusterName, clusterNamespace string) error {
```

Comment:
Please rename to `DeleteDashboardsDeployment`, the OS/OSD abbreviation is normally not used in the code.